src/main/scala/beam/analysis/cartraveltime/CarTripStatsFromPathTraversalEventHandler.scala

```diff
@@ -159,10 +164,19 @@ class CarTripStatsFromPathTraversalEventHandler(
       getIterationCarRideStats(event.getIteration, singleRideStats)
     }
 
+    iterationsCarTripInfo ++= type2Statistics.map {
+      case (carType, iterationCarTripStats) => (event.getIteration, carType) -> iterationCarTripStats
+    }
+
     averageCarSpeedPerIterationByType += type2Statistics.mapValues(_.speed.stats.avg)
 
+    //FIXME: create chart and store the image every time iteration ends... Why so?
     createRootGraphForAverageCarSpeedByType(event)
 
+    if (event.getIteration == lastIterationNumber) {
```

Comment:
I'd suggest to follow the same patter as for other graphs and re-create the graph on every iteration.

Reply:
Done